probe-rs/src/debug/snapshots/probe_rs__debug__debug_info__test__nRF52833_xxAA__full_unwind.snap

```diff
@@ -300,8 +300,8 @@ expression: stack_frames
                 - name:
                     Named: data_ptr
                   type_name:
-                    Pointer: ~
-                  value: "<referenced type> @ 0x20000040"
+                    Pointer: u8
```

Comment:
This is connected to my other pointer-related comment. `u8` is the type of the pointee, Rust does not store `DW_AT_type` for the pointer itself. GCC does not seem to store `DW_AT_type` for pointers at all, so there's a bit of a conflict here between the compilers - causing a bit of conflict in me, too :)

Reply:
Personally I prefer the way it used to be. The new way could easily be interpreted incorrectly as the pointer is a u8. The old way the user had to click one level deeper to see the type of the pointee, but at least there was no confusion.

Can you give an example of what the values look like for your C pointer so that I can take that into account in weighing pro's vs. con's?

Reply:
![image](https://github.com/probe-rs/probe-rs/assets/977627/86f34600-ffb3-4605-8730-0079b85415e9)
![image](https://github.com/probe-rs/probe-rs/assets/977627/6c21d41f-3f18-4239-80fa-53afabddac26)

Honestly, it looks about what I'd expect. What we can do, is that we can format the Rust pointer types a bit smarter - `if contains * { name } else { format!("*raw {name}") }` or with something like that - we have the machinery now :)

Reply:
That would be an improvement :)